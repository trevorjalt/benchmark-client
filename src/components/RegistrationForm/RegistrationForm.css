.Register__icon {
    height: 4vh;
    margin-right: 1vw;
}

.Register__button {
    display: flex;
    justify-content: center;
}

.RegisterSubmit {
    display: flex;
    align-items: center;
    margin: 4vh 0 4vh 0;

}

.RegistrationForm {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.Registration__prompt {
    color: #2FA7FB;
    font-size: 4vh;
}

.DemoContainer {
    display: flex;
    justify-content: center;
    /* align-items: center; */
}

.DemoInformation {
    background-color: #DAFDFD;
    width: 80vw;
    padding: 3vh;
    border-radius: 20px;
    margin: 1vh 0 6vh 0;
}

.bold {
    font-weight: bold;
}

.blue {
    color: #26547C;
}

@media only screen and (min-width: 800px) {
    .DemoInformation {
        width: 40vw;
    }
}