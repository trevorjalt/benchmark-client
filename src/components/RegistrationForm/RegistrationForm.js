import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'
import AuthApiService from '../../services/auth-api-service'
import RegisterIcon from '../Header/images/register-icon.png'
import './RegistrationForm.css'

export default class RegistrationForm extends Component {
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }

    state = { error: null }

    handleSubmit = event => {
        event.preventDefault()
        const { username, user_password, email, user_password_match } = event.target
        
        if (user_password.value !== user_password_match.value) {
            this.setState( {error: 'Passwords must match' })
            return
        } else {
       
        this.setState({ error: null })
        AuthApiService.postUser({
            username: username.value,
            user_password: user_password.value,
            email: email.value,
            // nickname: nickname.value,
        })
            .then(user => {
                username.value = ''
                user_password.value = ''
                email.value = ''
                user_password_match.value = ''
                this.props.onRegistrationSuccess()
            })
            .catch(res => {
                this.setState({ error: res.error })
            })
        }

    }
    render() {
        const { error } = this.state
        return (
            <div>
                <h3>Set your benchMark today</h3>
                <form 
                    className='RegistrationForm'
                    onSubmit={this.handleSubmit}
                >
                    <div role='alert'>
                        {error && <p className='red'>{error}</p>}
                    </div>
                    <div className='username'>
                        <label htmlFor="RegistrationForm__username">
                            Username <Required />
                        </label>
                        <Input
                            name='username'
                            type='text'
                            required
                            id='RegistrationForm_username'>
                        </Input>
                    </div>
                    <div className='email'>
                        <label htmlFor="RegistrationForm__email">
                            Email <Required />
                        </label>
                        <Input
                            name='email'
                            type='email'
                            required
                            id='RegistrationForm_email'>
                        </Input>
                    </div>
                    <div className='user_password'>
                        <label htmlFor="RegistrationForm__password">
                            Password <Required />
                        </label>
                        <Input
                            name='user_password'
                            type='password'
                            required
                            id='RegistrationForm__password'>
                        </Input>
                    </div>
                    <div className='user_password_match'>
                        <label htmlFor="RegistrationForm__password_match">
                            Re-Enter Password <Required />
                        </label>
                        <Input
                            name='user_password_match'
                            type='password'
                            required
                            id='RegistrationForm__password_match'>
                        </Input>
                    </div>
                    <div className='Register__button'>
                        <Button type='submit' className='RegisterSubmit'>
                        <img src={RegisterIcon} alt='Register' className='Register__icon' />
                            Register
                        </Button>
                    </div>
                </form>
            </div>

        )
    }
}