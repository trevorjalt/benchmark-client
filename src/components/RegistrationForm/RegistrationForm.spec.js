import React from 'react'
import RegistrationForm from './RegistrationForm'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe(`RegistrationForm Component`, () => {
    it (`renders without crashing`, () => {
        const wrapper = 
            shallow (
                <RegistrationForm />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})