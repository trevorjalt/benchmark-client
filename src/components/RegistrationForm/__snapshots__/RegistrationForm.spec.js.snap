// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm Component renders without crashing 1`] = `
<div>
  <h3>
    Set your benchMark today
  </h3>
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="username"
    >
      <label
        htmlFor="RegistrationForm__username"
      >
        Username 
        <Required />
      </label>
      <Input
        id="RegistrationForm_username"
        name="username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="email"
    >
      <label
        htmlFor="RegistrationForm__email"
      >
        Email 
        <Required />
      </label>
      <Input
        id="RegistrationForm_email"
        name="email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="user_password"
    >
      <label
        htmlFor="RegistrationForm__password"
      >
        Password 
        <Required />
      </label>
      <Input
        id="RegistrationForm__password"
        name="user_password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="user_password_match"
    >
      <label
        htmlFor="RegistrationForm__password_match"
      >
        Re-Enter Password 
        <Required />
      </label>
      <Input
        id="RegistrationForm__password_match"
        name="user_password_match"
        required={true}
        type="password"
      />
    </div>
    <div
      className="Register__button"
    >
      <Button
        className="RegisterSubmit"
        type="submit"
      >
        <img
          alt="Register"
          className="Register__icon"
          src="register-icon.png"
        />
        Register
      </Button>
    </div>
  </form>
</div>
`;
